rspec-mocks/lib/rspec/mocks/proxy.rb

```diff
@@ -31,13 +31,24 @@ def initialize(object, order_group, options={})
       end
 
       # @private
-      def ensure_can_be_proxied!(object)
-        return unless object.is_a?(Symbol)
+      def ensure_object_can_be_proxied!
+        return unless object_is_a_symbol?
```

Comment:
I don't know why this wasn't done at the time, but the traditional RSpec solution to this problem is to use `Class ===` because its much less likely to be overridden:
```suggestion
        return unless Symbol === object
```

Reply:
Hi, I thought about using `===` but was concerned about someone mocking `Class#===` (though I agree with you that this is a rare case).

I don't mind switching it to `===`, but wanted to double-check with you about the tradeoffs. Are we okay with accepting the slight risk that someone may stub or override `Class.#===` in order to get the benefit of cleaner / more idiomatic Ruby?

Reply:
If we're overly concerned, we can stash `===` and use the stashed one - we do this in a couple of places if I remeber correctly.

Reply:
I looked into the code, and there are other places (e.g., in `space.rb`) where we've got similar `Symbol#===` calls. These would require more time and risk to root out, so I think I'm okay with just doing `Symbol === object` in this PR as you suggest above.

22077c89fd8a67c3e0c23b19cf0cd31cd0bcdc55